src/coreclr/gc/gc.cpp

```diff
@@ -12845,8 +12847,23 @@ void gc_heap::distribute_free_regions()
         }
     }
 #else //MULTIPLE_HEAPS
-    while (decommit_step())
+    // we want to limit the amount of decommit we do per time to indirectly
```

Comment:
is similar logic already implemented for the svr case?

Reply:
Yes, in SVR case gc thread #0 wakes up every 100 ms and call decommit_step().